content/derived-datatypes.rst

```diff
@@ -350,8 +350,12 @@ sort of heterogeneous collection of basic datatypes recognized by MPI.
 .. figure:: img/E01-pack.svg
    :align: center
 
-   Schematic view of the relation between ``inbuf``, ``outbuf``, and the
-   ``position`` input/output parameter.
+   The relation of ``inbuf``, ``outbuf``, and ``position`` when calling
+   ``MPI_Pack``.  In this figure, ``outbuf`` already holds some data (the red
+   shaded area). The data in ``inbuf`` is copied to ``outbuf`` starting at the
+   address given with ``position``.  When the function returns, the ``position``
```

Comment:
Technically position is an offset into the buffer. An address is the value of a pointer.

```suggestion
   address ``outbuf+*position``.  When the function returns, the ``position``
```